### PR TITLE
Bump backstage to 1.30.4

### PR DESCRIPTION
**Pull request by ivangonzalezacuna**
First of all, the dependencies have been bumped to backstage 1.30.4.

Then, I've replaced some of the deprecated methods. Mostly caused by the `@backstage/backend-tasks` and `@backstage/backend-common` being deprecated and soon removed. There are still a bunch being used, but we will replace them in a follow-up PR (explanation in the last paragraph).

Finally, and as suggested in [this issue](https://github.com/backstage/community-plugins/issues/1176), the old backend system `createRouter` has to be deprecated. It will be removed completely in the next release.

Together with the removal of the old backend system, I would also suggest removing the standaloneServer completely (which has been broken for some time and it's being replaced by a new `example application` instead). Thanks to this, the last bunch of deprecations will be gone, as well as the unused packages. Some examples, which contain `packages` and possibly `examples` if some things are needed (like entity definitions, which is not needed for us):

* https://github.com/backstage/community-plugins/tree/main/workspaces/azure-devops
* https://github.com/backstage/community-plugins/tree/main/workspaces/azure-storage-explorer

PS: As a final task, I've merged the opened MRs to fix some of the security issues

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,5 +20,4 @@ jobs:
       - run: yarn install
       - run: yarn ci
         env:
-          CI: true
-      - run: yarn backstage-cli versions:check
```
Comment (heyLu) [inline .github/workflows/ci.yml:24]: This has no replacement, didn't the Backstage people suggest to replace this with something built into `yarn`?

File: plugins/s3-viewer-backend/src/service/router.ts [hunk 2]
```diff
@@ -35,6 +35,10 @@ import {
   PolicyDecision,
 } from '@backstage/plugin-permission-common';
 
+/**
+ * @deprecated Use the new backend system instead
```
Comment (heyLu) [inline plugins/s3-viewer-backend/src/service/router.ts:39]: Let's point people to https://github.com/spreadshirt/backstage-plugin-s3/tree/main/plugins/s3-viewer-backend for that.

File: .changeset/cool-rockets-confess.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+'@spreadshirt/backstage-plugin-s3-viewer-backend': patch
+---
+
+Replace some deprecations caused by the removal of the backend-tasks plugin and deprecate old backend system.
+Please, remove any usage of `RouterOptions` and `createRouter`.
```
Comment (heyLu) [inline .changeset/cool-rockets-confess.md:6]: Let's link https://github.com/spreadshirt/backstage-plugin-s3/tree/main/plugins/s3-viewer-backend here so people know where to go.

I think that README also should be adjusted to only mention the new backend.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ivangonzalezacuna) [reply .github/workflows/ci.yml:24]: Yes. The replacement is `yarn dedupe`. But this command is only available in yarn-3, and this repo is not using it. So there's no actual replacement unfortunately :( We could only bump yarn here and then use it

File: plugins/s3-viewer-backend/src/service/router.ts (hunk 2 above)
Comment (ivangonzalezacuna) [reply plugins/s3-viewer-backend/src/service/router.ts]: Done. Added the link in the 2 deprecation and also in the changeset

File: .changeset/cool-rockets-confess.md (hunk 3 above)
Comment (ivangonzalezacuna) [reply .changeset/cool-rockets-confess.md:6]: Added a note at the end. And also added a deprecation warning in the readme. The whole block will be removed together with the deprecated types. I think it's also good to do it this way if you like :)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (heyLu) [reply .github/workflows/ci.yml:24]: Ok, then let's try that later, could you comment this line out with a TODO instead so we maybe don't forget to add `yarn dedupe` later?

**Review by heyLu — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ivangonzalezacuna) [reply .github/workflows/ci.yml:24]: Did that :)

**Pull request merged**